lib/tpm2_options.c

```diff
@@ -424,13 +429,25 @@ tpm2_option_code tpm2_handle_options (int argc, char **argv,
 out:
 
     if (show_help) {
-        bool did_manpager = execute_man(argv[0]);
-        if (!did_manpager) {
+        bool did_manpager = true;
+        bool is_manpage = true;
+        bool is_short = true;
+        if (help_arg) {
+            is_manpage = strcmp(help_arg,"manpage");
+            is_short = strcmp(help_arg,"summary");
```

Comment:
isn't `--help=summary` and `--help-manpage` mutually exclusive?

You're already doing the strcmp on line 354, maybe just set a value for show_help and ditch help_arg:
0 - no help
1 - manpage
2 - summary

default:
show_help = 0

Set it:
show_help |= !strcmp(argv[optind],"manpage") << 0
show_help |= !strcmp(argv[optind],"summary") << 1